### ISSUE TITLE
 Instead of a string (the expected pk type), the provided value is: { name: 'stable' }

### ISSUE THREAD

**Issue by kolucciy** [ArekSredzki/electron-release-server#314]
Error **400** when running: `npm run publish`
Publish CMD: `cross-env-shell NODE_ENV=production electron-forge publish`

`"@electron-forge/publisher-electron-release-server": "^6.0.4",`

I'm not sure why a **string** is expected, but it looks like **channel** is passed as an object: https://github.com/electron/forge/blob/main/packages/publisher/electron-release-server/src/PublisherERS.ts#L100

### My Setup

**forge.config.js**
```
...
  publishers: [
    {
      name: '@electron-forge/publisher-electron-release-server',
      config: {
        baseUrl: 'https://myserver.com',
        username: 'username',
        password: 'password',
        flavor: 'kiosk'
      }
    }
  ],
...
```

**npm** logs:
```
[STARTED] [publisher-electron-release-server] Running the publish command
[FAILED] ERS publish failed with status code: 400 (https://myserver.com/api/version)
[FAILED] ERS publish failed with status code: 400 (https://myserver.com/api/version)

An unhandled rejection has occurred inside Forge:
Error: ERS publish failed with status code: 400 (https://myserver.com/api/version)
at fetchAndCheckStatus (C:\work\atm\node_modules\@electron-forge\publisher-electron-release-server\src\PublisherERS.ts:26:9)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at PublisherERS.publish (C:\work\atm\node_modules\@electron-forge\publisher-electron-release-server\src\PublisherERS.ts:97:9)
    at Task.task (C:\work\atm\node_modules\@electron-forge\core\src\api\publish.ts:108:15)
    at Task.run (C:\work\atm\node_modules\listr2\dist\index.cjs:978:11)
    at C:\work\atm\node_modules\p-map\index.js:57:22
node:events:491
      throw er; // Unhandled 'error' event
      ^

Error: spawn electron-forge ENOENT
    at notFoundError (C:\work\atm\node_modules\cross-spawn\lib\enoent.js:6:26)
    at verifyENOENT (C:\work\atm\node_modules\cross-spawn\lib\enoent.js:40:16)
    at ChildProcess.cp.emit (C:\work\atm\node_modules\cross-spawn\lib\enoent.js:27:25)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:291:12)
Emitted 'error' event on ChildProcess instance at:
    at ChildProcess.cp.emit (C:\work\atm\node_modules\cross-spawn\lib\enoent.js:30:37)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:291:12) {
  code: 'ENOENT',
  errno: 'ENOENT',
  syscall: 'spawn electron-forge',
  path: 'electron-forge',
  spawnargs: [ 'publish' ]
}

```
**docker container** logs:
```
verbo: Sending 400 ("Bad Request") response:
 UsageError: Invalid new record.
Details:
  Could not use specified `channel`.  Expecting an id representing the associated record, or `null` to indicate there will be no associated record.  But the specified value is not a valid `channel`.  Instead of a string (the expected pk type), the provided value is: { name: 'stable' }

    at /usr/src/electron-release-server/node_modules/sails/lib/hooks/blueprints/actions/create.js:89:11
    at /usr/src/electron-release-server/node_modules/sails/node_modules/async/dist/async.js:473:16
    at /usr/src/electron-release-server/node_modules/sails/node_modules/async/dist/async.js:2524:9
    at /usr/src/electron-release-server/node_modules/sails/node_modules/async/dist/async.js:473:16
    at replenish (/usr/src/electron-release-server/node_modules/sails/node_modules/async/dist/async.js:1009:25)
    at /usr/src/electron-release-server/node_modules/sails/node_modules/async/dist/async.js:1019:9
    at eachOfLimit (/usr/src/electron-release-server/node_modules/sails/node_modules/async/dist/async.js:1044:24)
    at /usr/src/electron-release-server/node_modules/sails/node_modules/async/dist/async.js:1049:16
    at Object.reduce (/usr/src/electron-release-server/node_modules/sails/node_modules/async/dist/async.js:2518:5)
    at Object.createRecord (/usr/src/electron-release-server/node_modules/sails/lib/hooks/blueprints/actions/create.js:36:9)
    at /usr/src/electron-release-server/node_modules/sails/lib/router/bind.js:248:46
    at routeTargetFnWrapper (/usr/src/electron-release-server/node_modules/sails/lib/router/bind.js:395:9)
    at Layer.handle [as handle_request] (/usr/src/electron-release-server/node_modules/express/lib/router/layer.js:95:5)
    at next (/usr/src/electron-release-server/node_modules/express/lib/router/route.js:137:13)
    at Route.dispatch (/usr/src/electron-release-server/node_modules/express/lib/router/route.js:112:3)
    at Layer.handle [as handle_request] (/usr/src/electron-release-server/node_modules/express/lib/router/layer.js:95:5)
    at /usr/src/electron-release-server/node_modules/express/lib/router/index.js:281:22
    at Function.process_params (/usr/src/electron-release-server/node_modules/express/lib/router/index.js:335:12)
    at next (/usr/src/electron-release-server/node_modules/express/lib/router/index.js:275:10)
    at next (/usr/src/electron-release-server/node_modules/express/lib/router/route.js:127:14)
    at /usr/src/electron-release-server/api/policies/authToken.js:35:5
    at /usr/src/electron-release-server/node_modules/jsonwebtoken/verify.js:223:12
```

**Comment by ArekSredzki** [comment]
Thanks for your ticket, this is due to a SailJS API change. We'll have to put up a PR to modify electron-forge to abide by the new API

**Comment by ArekSredzki** [comment]
I've put up a hotfix to the electron forge ERS plugin. You can try to use it locally to unblock yourself: https://github.com/electron/forge/pull/3149

**Comment by kolucciy** [comment]
Thanks a lot!!

**Comment by ArekSredzki** [comment]
This has now been fixed in Electron Forge [v6.0.5](https://github.com/electron/forge/releases/tag/v6.0.5)
